include/revng/ADT/KeyedObjectContainer.h

```diff
@@ -9,46 +9,61 @@
 
 #include "llvm/Support/YAMLTraits.h"
 
-#include "revng/ADT/KeyedObjectTraits.h"
 #include "revng/ADT/STLExtras.h"
 #include "revng/ADT/UpcastablePointer.h"
 #include "revng/Support/Assert.h"
 
 template<typename T>
-using KOTKey = decltype(KeyedObjectTraits<T>::key(std::declval<T>()));
+struct KeyedObjectTraits {
+  // static * key(const T &);
+  // static T fromKey(* Key);
```

Comment:
Drop, we have a concept for this.

Reply:
We still need to have at least a definition of this struct to be able to specialize it (which we do all over the place).

So I changed this declaration into a definition: https://github.com/revng/revng/pull/197/commits/8f6ce06ba692c3d7faa84361892396181d799560